### DISCUSSION TITLE
Executing WASM on the Data Host Chrome Extension

### DISCUSSION THREAD

**Discussion by frm** [trustfractal/protocol#21] [Q&A]
Manifest v3 [forbids execution of WASM outside a sandbox page](https://developer.chrome.com/docs/extensions/mv3/sandboxingEval/). This page doesn't have access to the Chrome APIs and can only communicate via `window.postMessage`. From my experience, it's a pain to set up, as a developer, and interact with, as a user.

If we're to provide an SDK or similar, this would mean that every extension would have to execute that same sandbox.

We're at a crossroads point that we can either support Manifest v2 with a clear API/SDK or handle all the pain of Manifest v3 and sandboxing.

My suggestion: I think going with Manifest v3 at this point leads to an unnecessary overhead in terms of development. We should downgrade to v2 and go with that, develop what we want to and show some concepts as soon as possible. Once we're at a stable point, we can upgrade to v3 and provide an abstraction to handle sandbox (who knows if it will even be needed by then).

What are your thoughts? @shelbyd @boymaas

**Comment by shelbyd** [comment]
That SGTM. From other things you said, it sounds like v3 isn't ready yet. I'd like to have as few requirements on our integration partners as possible, so if we can relax some that would be great.

Make sure to include a bit of a write-up somewhere about why we're using v2 instead of v3. Ideally with a checklist of our problems so we can track as they get resolved.

Perhaps make a GitHub issue for tracking upgrading to v3?

**Reply by frm** [reply]
A GitHub issue for tracking the upgrade seems perfect to me. I'll get on it.
